zospy/analyses/polarization.py

```diff
@@ -0,0 +1,344 @@
+    an_sett.ModifySettings(cfgoutfile, 'PPM_JY', str(jy))
+    an_sett.ModifySettings(cfgoutfile, 'PPM_PX', str(x_phase))
+    an_sett.ModifySettings(cfgoutfile, 'PPM_PY', str(y_phase))
+    an_sett.ModifySettings(cfgoutfile, 'PPM_WAVE', str(int(wavelength)))
+    an_sett.ModifySettings(cfgoutfile, 'PPM_FIELD', str(int(field)))
+    an_sett.ModifySettings(cfgoutfile, 'PPM_SURFACE', str(surface))
+    samp = utils.zputils.proc_constant(constants.Analysis.SampleSizes_ContrastLoss,
+                                 utils.zputils.standardize_sampling(sampling))
+    an_sett.ModifySettings(cfgoutfile, 'PPM_SAMP', str(samp-1))
+    an_sett.ModifySettings(cfgoutfile, 'PPM_ADDCONFIG', str(add_configs))
+    an_sett.ModifySettings(cfgoutfile, 'PPM_SUBCONFIGS', str(sub_configs))
+
+    an_sett.LoadFrom(cfgoutfile)
+
+    # Run analysis
+    analysis.ApplyAndWaitForCompletion()
+
+    # Get results
+    analysis.Results.GetTextFile(txtoutfile)
+    line_list = [line for line in open(txtoutfile, 'r', encoding='utf-16-le')]
+
+    # Create output dict
+    data = AttrDict()
+
+    # Add header
+    hdr = line_list[0].strip().replace('\ufeff', '')
+    data['Header'] = hdr
+
+    # Add settings
+    for ii in range(9):
+        name = line_list[7+ii].replace(' ','').split(':')[0]
+        data[name] = float(re.sub(r'[^\d\.]+', '', line_list[7+ii]))
+
+    # Read data table as dataframe
+    df = pd.read_csv(StringIO(''.join(line_list[17:]).replace(' ','')),
+                     delimiter='\t', decimal=_config.DECIMAL)
+
+    # Add to result dictionary
+    data['Table'] = df
+
+    # Get headerdata, metadata and messages
+    headerdata = utils.zputils.analysis_get_headerdata(analysis)
+    metadata = utils.zputils.analysis_get_metadata(analysis)
+    messages = utils.zputils.analysis_get_messages(analysis)
+
+    # Get settings
+    settings = pd.Series(name='Settings',dtype=object)
+
+    settings.loc['Jx'] = jx
+    settings.loc['Jy'] = jy
+    settings.loc['X-Phase'] = x_phase
+    settings.loc['Y-Phase'] = y_phase
+    settings.loc['Wavelength'] = int(wavelength)
+    settings.loc['Field'] = int(field)
+    settings.loc['Surface'] = surface
+    settings.loc['Sampling'] = sampling
+    settings.loc['Add Configs'] = add_configs
+    settings.loc['Sub Configs'] = sub_configs
+
+    # Create output
+    ret = AnalysisResult(analysistype=analysistype, data=data, settings=settings, metadata=metadata,
+                         headerdata=headerdata, messages=messages,
+                         RawTextData=line_list, CgfOutFile=cfgoutfile, TxtOutFile=txtoutfile)
+
+    # cleanup if needed
+    if cleancfg:
+        os.remove(cfgoutfile)
+    if cleantxt:
+        os.remove(txtoutfile)
+
+    # Process oncomplete
+    process_oncomplete(analysis=analysis, oncomplete=oncomplete, ret=ret)
+
+    return ret
+
+
+
+
+def transmission(oss, sampling='32x32', unpolarized=False, jx=1, jy=0, x_phase=0, y_phase=0,

```

Comment:
All comments from the previous analysis apply for this one as well

Reply:
Understood. Note that here modifying the settings file was very cumbersome. I think Zemax forgot to implement some functions so I had to reverse-engineer the byte arrays to figure out how to correctly change them. But I tested it and it works well now.